### DISCUSSION TITLE
Upgrade from Mirth Connect 3.4 to 4.1

### DISCUSSION THREAD

**Discussion by nehalpatelus** [nextgenhealthcare/connect#5386] [Q&A] [answered]
Hi
We are using Mirth connect 3.4 in the VM with MySQL 5.7.  We are planning to upgrade or migrate to latest version using Docker image .
My questions are .
1-  Can we use our old DB backup with new DB server ?

**Comment by pacmano1** [comment, marked as answer]
I would upgrade your current install first.   Then move to Docker, adjusting the DB config for docker to point to your mysql instance.

**Reply by nehalpatelus** [reply]
Thanks for reply . I can directly update from 3.4 version to 4,1  ?
will that new version make necessary changes in to DB ?

**Reply by pacmano1** [reply]
You should read the upgrade guides for each release.   "directly upgrade?"   Maybe, depends on your Mirth code.

**Reply by nehalpatelus** [reply]
I mean to say .. i need to update - 3.4 then 3.5,  3.6 .... 4.1 
or 3.4 > 4.1 I can apply ?

**Reply by pacmano1** [reply]
No, you do not need to do that.   But like any upgrade, you should test it all out not in production.

**Reply by nehalpatelus** [reply]
If i update my current node . it would be riskily if somehow it failed 

can I do this way 
Create a new DB Instance and restore the DB in it 
Create docker container with 3.4 version  and point it out to it 
upgrade the Docker image to 4.1 and check ?

**Reply by nehalpatelus** [reply]
> No, you do not need to do tha

.. No for what ? 3.4 > 4.1 , you mean to say i need to install the version one by one and then test ?

**Reply by pacmano1** [reply]
You do not need to install versions one by one.

I already told you what I would do which is upgrade the current engine in place before you jump to Docker.   After of course you tried an upgrade in in your test/lab environment.   That's mostly a tried and true system administrator tenet - "don't change too much at once".

As for other options, either try them or someone else will chime in.   

Again, you need to read the upgrade guides and see how they may affect your mirth code.

**Reply by nehalpatelus** [reply]
ok . Thanks for your help

**Reply by prabeshdahal1991** [reply]
@nehalpatelus one more thing before you upgrade. Please follow this link
https://github.com/nextgenhealthcare/connect/wiki/3.7.0---Upgrade-Guide

**Reply by prabeshdahal1991** [reply]
In this forum, it says that from Mirth version 3.7, commercial extensions now require a license key to be set in mirth.properties
So, if there are commercial extensions installed in version 3.4, updating to 4.1.0 requires a license key.
If there are no commercial extensions, then no need of license key.

**Comment by tonygermano** [comment]
The official mirth docker image does not include a database (besides embedded derby.) You can run mysql in a separate docker container and restore to it before turning on mirth and pointing to that database.
